packages/docregistry/src/default.ts

```diff
@@ -1,40 +1,64 @@
 // Copyright (c) Jupyter Development Team.
 // Distributed under the terms of the Modified BSD License.
 
+import { MainAreaWidget } from '@jupyterlab/apputils';
+
+import {
+  CodeEditor,
+  CodeEditorData,
+  ICodeEditorData
+} from '@jupyterlab/codeeditor';
+
 import { Mode } from '@jupyterlab/codemirror';
 
+import { IChangedArgs, PathExt } from '@jupyterlab/coreutils';
+
+import { DatastoreExt } from '@jupyterlab/datastore';
+
 import { Contents } from '@jupyterlab/services';
 
 import { JSONValue } from '@phosphor/coreutils';
 
+import { IDisposable } from '@phosphor/disposable';
+
 import { ISignal, Signal } from '@phosphor/signaling';
 
 import { Widget } from '@phosphor/widgets';
 
-import { MainAreaWidget } from '@jupyterlab/apputils';
-
-import { CodeEditor } from '@jupyterlab/codeeditor';
-
-import { IChangedArgs, PathExt } from '@jupyterlab/coreutils';
-
-import { IModelDB } from '@jupyterlab/observables';
-
 import { DocumentRegistry, IDocumentWidget } from './index';
 
 /**
- * The default implementation of a document model.
+ * The default implementation of a text document model.
  */
-export class DocumentModel extends CodeEditor.Model
+export class TextDocumentModel extends CodeEditor.Model
   implements DocumentRegistry.ICodeModel {
   /**
    * Construct a new document model.
    */
-  constructor(languagePreference?: string, modelDB?: IModelDB) {
-    super({ modelDB });
-    this._defaultLang = languagePreference || '';
-    this.datastore.changed.connect(this.triggerContentChange, this);
+  constructor(options: TextDocumentModel.IOptions = {}) {
```

Comment:
The default document model is now a collaborative `TextDocumentModel`, which stores its string data in a datastore. There is also a simpler `StringDocumentModel`, which is intended for large blobs of text which are not collaborative.